src/Go/boardAnalysis/goAI.ts

```diff
@@ -15,20 +15,29 @@ import {
   getAllEyesByChainId,
   getAllNeighboringChains,
   getAllValidMoves,
+  getPreviousMoveDetails,
 } from "./boardAnalysis";
 import { findDisputedTerritory } from "./controlledTerritory";
 import { findAnyMatchedPatterns } from "./patternMatching";
 import { WHRNG } from "../../Casino/RNG";
 import { Go, GoEvents } from "../Go";
 
 let currentAITurn: Promise<Play> | null = null;
+let currentTurnResolver: ((value: Play | PromiseLike<Play>) => void) | null = null;
 
 /**
  * Retrieves a move from the current faction in response to the player's move
  */
 export function makeAIMove(boardState: BoardState): Promise<Play> {
   // If AI is already taking their turn, return the existing turn.
-  if (currentAITurn) return currentAITurn;
+  if (currentAITurn) {
```

Comment:
I think this is the only thing that reads `currentAITurn` anymore. Probably it should be turned back into a boolean named `isAIThinking` or smth.